pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -1537,6 +1537,12 @@ type ObjectStoreUserSpec struct {
 	// +optional
 	// +nullable
 	Quotas *ObjectUserQuotaSpec `json:"quotas,omitempty"`
+	// +optional
+	// +nullable
+	AccessKey string `json:"accessKey,omitempty"`
+	// +optional
+	// +nullable
+	SecretKey string `json:"secretKey,omitempty"`
```

Comment:
While this is very simple, it will also put the secret in plain text in the CR. A more common approach is to reference a secret name, and read the secret value. However, secrets still aren't very secret since they're just base64 encoded. So we need to understand scenarios when this would be used.
```suggestion
	SecretKeyRef string `json:"secretKey,omitempty"`
```

Reply:
But now that I'm thinking about this more, I think we need a different approach. The current implementation for the object user will generate the user access key and secret key, then store them in a k8s secret as seen [here](https://github.com/rook/rook/blob/2965373ea9b45a475222a84c17fd4fb9008ffb03/pkg/operator/ceph/object/user/controller.go#L427-L459). The user could just create that secret before creating the CR, then the operator could use it instead of generating a new secret. And either we could expect them to create the secret with a name that already matches the current implementation, or we could have the `secretKeyRef` field where they could override the default name.

Reply:
@travisn good suggestion, maybe we need to assigin the name and namespace of  the secret in the cr, like this: https://kubernetes-csi.github.io/docs/secrets-and-credentials-storage-class.html,  if the secret was assigned in the cr, the operator no need to generate new secret.

Reply:
"ceph.rook.io/user-secret-name" and "ceph.rook.io/user-secret-namespace" is ok?

Reply:
if we want to make it easier，I thinks no need to use secret and configmaps， there are too many related things to deal with, base64 in the cr is more simple and more convenient

Reply:
The easiest thing would be to use the existing secret that rook creates currently with the creds. Rook would look if that secret already exists, then load that secret and create the object user with those creds. Then no need even for a new setting in the user CRD.

Reply:
I have 2 question:
1. If the existing secret not under the namespace "rook-ceph", we don't know where to get the secret.
2. the "secretName" under status,  if not invalid to be assign by user.

Reply:
The current secret created by rook is only in the same namespace as the ceph cluster. So if we need to give flexibility for the user to define the name and namespace of the secret to store the creds, then we would need a secretRef field in the CR.

Reply:
Because of if we need to create the secret pre,  we can not limit user to create the secret in a special namespace, so we need to a secretRef and the namespace of the secret in the cr.

Reply:
How about if a secretRef defines a secret in a different namespace, we expect it to already exist? Then rook doesn't need access to write secrets to another namespace. Although we can double check, rook may already have that access for OBCs.

Reply:
we need to create the secret manually, so we can not limit user to create the secret in special namespace

Reply:
Can you expand on the scenario? What really is the use case? This is starting to sound more like an OBC implementation. With an OBC, a user will be created and the bucket creds stored in the namespace of the OBC with accerss to that bucket. Do you not want to use an OBC perhaps because the user creds only work with a single bucket or because you can't define the creds yourself?